### PR TITLE
Remove for..of iterators

### PR DESCRIPTION
**Pull request by chadhietala**

### CODE REVIEW

File: packages/@glimmer/runtime/lib/scanner.ts [hunk 1]
```diff
@@ -58,9 +60,10 @@ export class InlineBlock extends Template {
       builder.bindPositionalArgsForLocals(locals);
     }
 
-    for (let statement of this.statements) {
+    for (let i = 0; i < this.statements.length; i++) {
+      let statement = this.statements[i];
       let refined = SPECIALIZE.specialize(statement, table);
-      STATEMENTS.compile(refined, builder);
```
Comment (rwjblue) [inline packages/@glimmer/runtime/lib/scanner.ts]: So this was just wrong before, right?

Comment (rwjblue) [inline packages/@glimmer/runtime/lib/scanner.ts]: Hmm, no it seems like there is no `b` in this scope, maybe it needs to be reverted?

**Comment by Turbo87** [comment]
@chadhietala why did you remove for-of?

**Comment by locks** [comment]
@Turbo87 performance concerns, and the fact that it falls back to iterables

**Pull request merged**